yourapp/gateway/main.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"context"
+	"log"
+	"net/http"
+
+	"github.com/temporalio/samples-go/yourapp"
+
+	"go.temporal.io/sdk/client"
+)
+
+func main() {
+	// Create a Temporal Client to communicate with the Temporal Cluster.
+	// A Temporal Client is a heavyweight object that should be created just once per process.
+	temporalClient, err := client.NewClient(client.Options{
+		HostPort: client.DefaultHostPort,
+	})
+	if err != nil {
+		log.Fatalln("Unable to create Temporal Client", err)
+	}
+	defer temporalClient.Close()
+	// Start an HTTP server and listen on /start
+	http.HandleFunc("/start", func(w http.ResponseWriter, r *http.Request) {
+		startWorkflowHandler(w, r, temporalClient)
+	})
+	http.ListenAndServe(":8081", nil)
```

Comment:
We actually hear a lot from newer users that seeing the temporal client used like this immediately helps to orient themselves around the model of the system.

Seems like pretty low overhead to accommodate wdyt?

And users can start WFs with curl commands rather than running a main.go file (yes it is 1 more step) 
But still, the value add for newer users could be quite big I hear.

Reply:
Most Go developers can also write a CLI, but we include that because that's not the point - the point is showing how to start a workflow execution from code. What we need to show is specifically and explicitly HOW a developer uses Temporal in the context of a different kind of application. This comes up so often. I can assure you it's not as obvious to people new to Temporal.

Reply:
> What we need to show is specifically and explicitly HOW a developer uses Temporal in the context of a different kind of application.

Then make that the focal point. No need to litter _basic_ samples with HTTP servers. How am I supposed to know that the `/yourapp` sample shows me how to call Temporal from an HTTP server?

Reply:
> Then make that the focal point. No need to litter basic samples with HTTP servers

Agree here. We should have *an* example of this pattern.